ddl/dm_schema_tracker.go

```diff
@@ -0,0 +1,686 @@
+// Copyright 2022 PingCAP, Inc.
```

Comment:
I think we can create a separate directory for this file under the `ddl` directory. 
And we can move `info_store.go` to this dir too(this puts all the relevant code together).

Reply:
> And we can move info_store.go to this dir too(this puts all the relevant code together).

@hawkingrei says it may affect CI speed (in Lark chatting). 

> I think we can create a separate directory for this file under the ddl directory.

so it will be located in another package, maybe it will cause cycle import. I'll try it soon in this PR.

Reply:
There's a cycle:

My SchemaTracker should implement `DDL` interface, so it should import `pingcap/tidb/ddl`

And I reuses `TestChangingDBCharset` in `ddl/db_integration_test.go` to test SchemaTracker, so `pingcap/tidb/ddl` should import the package of SchemaTracker.

I can move DDL interface into a separate package if you agree.

Reply:
also cc @hawkingrei

Reply:
In the origin PR, I just see the [dm_schema_tracker.go](https://github.com/pingcap/tidb/pull/35991/files#diff-99996cb5925a5a1316d5ea1edf8e95240abe0335fce8646c202f96769fce01bc) that have not relation with ddl
 components. if it is related with ddl, it can be put at the ddl.

Reply:
but We should put them into different folder as soon as possible.

Reply:
> I can move DDL interface into a separate package if you agree.

I agree with you.

Reply:
@tangenta @xiongjiwei @wjhuang2016 PTAL